include/ui_components/elements/text_renderer.hpp

```
@@ -14,13 +15,32 @@ class TextRenderer : public UIComponent {
+  void layout(UIConstraints size) override;
   void draw(SkCanvas *canvas) override;
 
  protected:
   void debugFillProperties(std::ostringstream &os, int indent) const override;
 
  private:
   std::string text_;
+  const TextStyle &style_;
   ParagraphBuilder paragraphBuilder_;
+
+  uint64_t layoutHashCode() const noexcept {
```

Clang (14.0.6)

[warning] function 'layoutHashCode' should be marked [[nodiscard]]

(modernize-use-nodiscard)